src/App.tsx

```diff
@@ -8,7 +12,81 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+type ActiveModal = {
+  todo: Todo | null,
+  index: number,
+  isActive: boolean,
+};
+
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [isActiveTodoList, setIsActiveTodoList] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+  const [activeModal, setActiveModal] = useState<ActiveModal>({
+    todo: null,
+    index: 0,
+    isActive: false,
+  });
```

Comment:
initial state for selectedTodo can be null
suggest to name this state selectedTodo